### ISSUE TITLE
Blenderbot output logits dimensions mismatch

### ISSUE THREAD

**Issue by naderabdalghani** [huggingface/transformers#12744]
- `transformers` version: 4.9.0.dev0
- Platform: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.10
- PyTorch version (GPU?): 1.9.0+cu102 (True)
- Tensorflow version (GPU?): 2.5.0 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

Models: Blenderbot — @patrickvonplaten, @patil-suraj

Documentation: @sgugger

## Information

The model I am using is `facebook/blenderbot-400M-distill`.

The problem arises when using:
* [x] my own modified scripts

## To reproduce

Steps to reproduce the behaviour:

Run the following code
```python
from transformers import BlenderbotTokenizer, BlenderbotForConditionalGeneration
import torch

DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")

model = BlenderbotForConditionalGeneration.from_pretrained('facebook/blenderbot-400M-distill').to(DEVICE)
tokenizer = BlenderbotTokenizer.from_pretrained('facebook/blenderbot-400M-distill')

input_ids = tokenizer.encode("Hello there! My name is Nader", return_tensors="pt").to(DEVICE)
decoder_input_ids = tokenizer.encode("<s>", return_tensors="pt").to(DEVICE)
outputs = model(input_ids=input_ids, decoder_input_ids=decoder_input_ids)

logits = outputs.logits

print(input_ids.shape)
print(logits.shape)
```
## Current output

```
torch.Size([1, 9])
torch.Size([1, 3, 8008])
```

## Expected behaviour

```
torch.Size([1, 9])
torch.Size([1, 9, 8008])
```

According to [the documentation of the `forward()` method,](https://huggingface.co/transformers/model_doc/blenderbot.html?highlight=forward#transformers.BlenderbotForConditionalGeneration.forward) it should return a `Seq2SeqLMOutput` object with a member `logits` having the following properties 

> logits (`torch.FloatTensor` of shape (`batch_size`, `sequence_length`, `config.vocab_size`))

and considering that the `input_ids` in this case has dimensions of `(batch_size, sequence_length)` which maps to `(1, 9)` in this case shown in code above, why doesn't the model output logits have dimensions of `(1, 9, 8008)`? Please tell me what does this `3` signify if I'm missing something here. Thanks.

**Comment by naderabdalghani** [comment]
Seems like a documentation thing. After a little bit of digging, I found out that the output logits have the dimensions of `(batch_size, decoder_sequence_length, config.vocab_size)` not `(batch_size, sequence_length, config.vocab_size)`.

**Comment by patrickvonplaten** [comment]
@naderabdalghani thanks for investigating! Also feel free to open a PR to clarify the docs if you'd like :-)
